pkg/querier/parquet_queryable.go

```diff
@@ -0,0 +1,346 @@
+package querier
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus-community/parquet-common/schema"
+	"github.com/prometheus-community/parquet-common/search"
+	parquet_storage "github.com/prometheus-community/parquet-common/storage"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+	"github.com/prometheus/prometheus/util/annotations"
+	"github.com/thanos-io/thanos/pkg/strutil"
+
+	"github.com/cortexproject/cortex/pkg/storage/bucket"
+	cortex_tsdb "github.com/cortexproject/cortex/pkg/storage/tsdb"
+	"github.com/cortexproject/cortex/pkg/storage/tsdb/bucketindex"
+	"github.com/cortexproject/cortex/pkg/tenant"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/cortexproject/cortex/pkg/util/multierror"
+	"github.com/cortexproject/cortex/pkg/util/services"
+)
+
+type parquetQueryableFallbackMetrics struct {
+	blocksQueriedTotal *prometheus.CounterVec
+}
+
+func newParquetQueryableFallbackMetrics(reg prometheus.Registerer) *parquetQueryableFallbackMetrics {
+	return &parquetQueryableFallbackMetrics{
+		blocksQueriedTotal: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Name: "cortex_parquet_queryable_blocks_queried_total",
+			Help: "Total number of blocks found to query.",
+		}, []string{"type"}),
+	}
```

Comment:
Should we add some other metrics to look at this at query level.
Number of queries that query 100% TSDB blocks, 100% parquet blocks and touch both?

Reply:
I can only do at the "Select" level.

I guess that what u meant, right?

Reply:
Done!